langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/ContentsMapper.java

```diff
@@ -2,39 +2,100 @@
 
 import com.google.cloud.vertexai.api.Content;
 import com.google.cloud.vertexai.api.Part;
+import dev.langchain4j.data.message.AiMessage;
 import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageType;
 import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 class ContentsMapper {
     static class InstructionAndContent {
         public Content systemInstruction = null;
         public List<Content> contents = new ArrayList<>();
+
+        @Override
+        public String toString() {
+            return "InstructionAndContent {\n" +
+                " systemInstruction = " + systemInstruction +
+                ",\n contents = " + contents +
+                "\n}";
+        }
     }
 
     static InstructionAndContent splitInstructionAndContent(List<ChatMessage> messages) {
         InstructionAndContent instructionAndContent = new InstructionAndContent();
-
         List<Part> sysInstructionParts = new ArrayList<>();
 
-        for (ChatMessage message : messages) {
-            if (message instanceof SystemMessage) {
-                sysInstructionParts.addAll(PartsMapper.map(message));
+        List<ToolExecutionResultMessage> executionResultMessages = new ArrayList<>();
+
+        for (int msgIdx = 0; msgIdx < messages.size(); msgIdx++) {
```

Comment:
the logic of this method is quite complicated, could you please cover it with unit tests?

Reply:
Good idea. It was a bit tricky to get right. Adding some tests.